### ISSUE TITLE
ReadOnlySpan<T> ToString() leads to System.InvalidProgramException

### ISSUE THREAD

**Issue by mharthoorn** [dotnet/coreclr#15458]
Tried out System.Memory package
https://www.nuget.org/packages/System.Memory/4.4.0-preview1-25305-02

The use of ReadOnlySpan.ToString() compiles correctly, but leads to an invalid program. 

*In Visual Studio 15.5:*
"Your app has entered a break state, but no code is currently executing that is supported by the   
selected debug engine (e.g. only native runtime code is executing)."

*dotnet cli:*
"Unhandled Exception: System.InvalidProgramException: Common Language Runtime detected an invalid program. at SpanOfTest.Program.Main(String[] args)"

Minimal example that leads to this error
```csharp
  static void Main(string[] args)
  {
      var span = "Hello world!".AsSpan();
      Console.WriteLine(span.ToString());
  }
```
(Not sure if this is the right place to post this)

**Comment by jkotas** [comment]
cc @ahsonkhan

**Comment by ahsonkhan** [comment]
Duplicate of https://github.com/dotnet/corefx/issues/21629?

**Comment by jkotas** [comment]
Fixed by #15458

**Comment by ahsonkhan** [comment]
This is the PR where the issue got fixed: https://github.com/dotnet/coreclr/pull/15745
